code/game/machinery/computer/communications.dm

```diff
@@ -116,6 +116,19 @@ var/const/CALL_SHUTTLE_REASON_LENGTH = 12
 			else if (src.authenticated==2 && message_cooldown)
 				usr << "Intercomms recharging. Please stand by."
 
+		if("crossserver")
+			if(src.authenticated==2)
+				if(CM.lastTimeUsed + 600 > world.time)
+					usr << "Arrays recycling.  Please stand by."
+					return
+				var/input = stripped_input(usr, "Please choose a message to transmit to an allied station.  Please be aware that this process is very expensive, and abuse will lead to... termination.", "Send a message to an allied station.", "")

```

Comment:
You probably want to use stripped_multiline_input as it's easier to write long messages with.